packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -64,7 +64,16 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   // We set `autoUncompress` to false to ensure that we can trust the value of
   // the `Content-Length` HTTP header. We automatically uncompress the content
   // in our call to [consolidateHttpClientResponseBytes].
-  static final HttpClient _sharedHttpClient = HttpClient()..autoUncompress = false;
+  //
+  // In order to allow HttpOverrides to be used in different parts of a Flutter
+  // application, use these or the parent zone as a key to look up the client.
+  static final Expando<HttpClient> _sharedClients = Expando<HttpClient>();
+  static HttpClient get _sharedHttpClient {
+    final Object token = HttpOverrides.current ?? Zone.root;
+    HttpClient? cachedClient = _sharedClients[token];
+    cachedClient ??= (_sharedClients[token] = HttpClient()..autoUncompress = false);
+    return cachedClient;
```

Comment:
What happens if your application decides to change the http overrides for the current zone multiple times? 

IOW, why wouldn't we _always_ check for HttpOverrides.current and just avoid caching the client altogether? 

Also, it seems like we should be asserting that `autoUncompress` is false on whatever client we get from the overrides, otherwise it'll break the loading stuff unexpectedly.

Reply:
hmmm, that's a good point. let me check what the current behavior is if we don't set autocompress. Not caching if there is an http override seems like a better change too

Reply:
I just like using expandos

Reply:
If autoUncompress is set to true, we get a content-length that doesn't match the size of the bytestream we get or something, and it throws off the callbacks for how much progress we've made loading.